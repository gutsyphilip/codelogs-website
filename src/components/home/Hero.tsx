import AvatarImage from '../../assets/images/avatar.jpg'
import { Button } from '../utility/button'
import { FeaturedCard } from '../utility/cards'
import React from 'react'
import { cardVariantTypes } from '../utility/cards/card.type'
import styles from './home.module.scss'

const Hero = () => {
  return (
    <section className={styles.hero}>
      <div className={styles.hero_caption}>
        <h2 className={styles.hero_caption_title}>Welcome to Codelogs!</h2>
        <p className={styles.hero_caption_description}>
          Quality streams on advanced software engineering concepts taught by
          experienced professionals.
        </p>
        <Button className={styles.hero_caption_button} variant="themed">
          Watch videos
        </Button>
      </div>

      <div>
        <FeaturedCard
          className={styles.hero_upNext}
          data={{
            date: 'Saturday, May 2 2020 - 18:00 WAT',
            label: 'COMING SOON',
            title: 'Defining your Testing Strategy using the Testing Pyramid.',
            image: AvatarImage
          }}
          hasPadding
          variant={cardVariantTypes.light}
        />
      </div>
    </section>
  )
}

export default Hero
