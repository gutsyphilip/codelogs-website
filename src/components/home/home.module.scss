@import "../../styles/breakpoint";
@import "../../styles/converter";
@import "../../styles/variables";

.hero {
  // margin-top: to-rem(53);

  @include md() {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
  }
  &_caption {
    margin-top: to-rem(63);
    flex: 1;
    width: 100%;

    @include md() {
      min-width: to-rem(400);
    }

    &_title {
      font-size: to-rem(28);
      color: $theme-purple;
    }

    &_description {
      font-size: to-rem(18);
      color: $text-light;
      margin-top: to-rem(13);
    }

    &_button {
      margin-top: to-rem(24);
    }
  }

  &_upNext {
    margin-top: to-rem(82);
    @include lg() {
      margin-left: to-rem(20);
      max-width: to-rem(383);
      width: to-rem(383);
    }
  }
}
