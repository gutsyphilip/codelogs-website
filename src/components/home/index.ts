export * from './Hero'
export * from './Episodes'
