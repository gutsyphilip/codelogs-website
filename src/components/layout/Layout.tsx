/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import { graphql, useStaticQuery } from 'gatsby'

import Header from '../header/Header'
import PropTypes from 'prop-types'
import React from 'react'
import styles from './layout.module.scss'

const Layout: React.FC = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return (
    <div className={styles.layout}>
      <Header />
      <div>
        <main>{children}</main>
        {/* <footer>
          Â© {new Date().getFullYear()}, Built with
          {` `}
          <a href="https://www.gatsbyjs.org">Gatsby</a>
        </footer> */}
      </div>
    </div>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired
}

export default Layout
