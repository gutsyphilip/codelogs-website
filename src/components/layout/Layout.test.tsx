import React from 'react'
import Layout from './Layout'
import { Hero } from '../home'
import renderer from 'react-test-renderer'

describe('Layout', () => {
    it('renders correctly', () => {
        const tree = renderer.
        create(<Layout> <Hero/> </Layout>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
