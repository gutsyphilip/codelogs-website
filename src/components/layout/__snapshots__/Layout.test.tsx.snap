// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="layout"
>
  <header>
    <a
      href="/"
    >
      <test-file-stub />
       
      
    </a>
  </header>
  <div>
    <main>
       
      <section
        className="hero"
      >
        <div
          className="hero_caption"
        >
          <h2
            className="hero_caption_title"
          >
            Welcome to Codelogs!
          </h2>
          <p
            className="hero_caption_description"
          >
            Quality streams on advanced software engineering concepts taught by experienced professionals.
          </p>
          <button
            className="button hero_caption_button button_themed"
          >
            Watch videos
          </button>
        </div>
        <div>
          <section
            className="card hero_upNext featuredCard card_light card_hasPadding"
          >
            <div>
              <h6
                className="featuredCard_label"
              >
                COMING SOON
              </h6>
              <h2
                className="featuredCard_title"
              >
                Defining your Testing Strategy using the Testing Pyramid.
              </h2>
              <p
                className="featuredCard_date"
              >
                Saturday, May 2 2020 - 18:00 WAT
              </p>
            </div>
            <div
              className="featuredCard_speaker"
            >
              <img
                alt=""
                className="featuredCard_speaker_image"
                src="test-file-stub"
              />
              <div
                className="featuredCard_speaker_details"
              >
                <h4
                  className="featuredCard_speaker_details_title"
                >
                  Philip Obosi
                </h4>
                <h6
                  className="featuredCard_speaker_details_position"
                >
                  Snr. Software Engineer
                </h6>
                <p
                  className="featuredCard_speaker_details_company"
                >
                  Kudi Inc.
                </p>
              </div>
            </div>
          </section>
        </div>
      </section>
       
    </main>
  </div>
</div>
`;
